From: NeilBrown <neilb@suse.de>
Date: Thu, 12 Apr 2012 16:27:11 +1000
Subject: [PATCH 21/23] md/bitmap: make sure reshape request are reflected in
 superblock.3.5
Patch-mainline:fate#311379 
References: 

As a reshape may change the sync_size and/or chunk_size, we need
to update these whenever we write out the bitmap superblock.

Signed-off-by: NeilBrown <neilb@suse.de>
Acked-by: NeilBrown <neilb@suse.de>

---
 drivers/md/bitmap.c |    3 +++
 1 file changed, 3 insertions(+)

--- linux-3.0-SLE11-SP2-BTMU.orig/drivers/md/bitmap.c
+++ linux-3.0-SLE11-SP2-BTMU/drivers/md/bitmap.c
@@ -476,6 +476,9 @@ void bitmap_update_sb(struct bitmap *bit
 	/* Just in case these have been changed via sysfs: */
 	sb->daemon_sleep = cpu_to_le32(bitmap->mddev->bitmap_info.daemon_sleep/HZ);
 	sb->write_behind = cpu_to_le32(bitmap->mddev->bitmap_info.max_write_behind);
+	/* This might have been changed by a reshape */
+	sb->sync_size = cpu_to_le64(bitmap->mddev->resync_max_sectors);
+	sb->chunksize = cpu_to_le32(bitmap->mddev->bitmap_info.chunksize);
 	kunmap_atomic(sb, KM_USER0);
 	write_page(bitmap, bitmap->storage.sb_page, 1);
 }
